server/src/server/keyspace/KeyspaceManager.java

```diff
@@ -18,149 +18,29 @@
-        this.existingKeyspaces = ConcurrentHashMap.newKeySet();
-    }
-
-    /**
-     * Add new keyspace in graknsystem if it does not exist already.
-     *
-     * @param keyspace The new KeyspaceImpl we have just created
-     */
-    public void putKeyspace(KeyspaceImpl keyspace) {
-        if (containsKeyspace(keyspace)) return;
-
-        try (TransactionOLTP tx = systemKeyspaceSession.transaction().write()) {
-            AttributeType<String> keyspaceName = tx.getSchemaConcept(KEYSPACE_RESOURCE);
-            if (keyspaceName == null) {
-                throw GraknServerException.initializationException(keyspace);
-            }
-            Attribute<String> attribute = keyspaceName.create(keyspace.name());
-            tx.<EntityType>getSchemaConcept(KEYSPACE_ENTITY).create().has(attribute);
-            tx.commit();
-
-            // add to cache
-            existingKeyspaces.add(keyspace);
-        } catch (InvalidKBException e) {
-            throw new RuntimeException("Could not add keyspace [" + keyspace + "] to system graph", e);
-        }
-    }
-
-    public void closeStore() {
-        this.systemKeyspaceSession.close();
-        this.graph.close();
-    }
-
-    private boolean containsKeyspace(KeyspaceImpl keyspace) {
-        //Check local cache
-        if (existingKeyspaces.contains(keyspace)) {
-            return true;
-        }
-
-        try (TransactionOLTP tx = systemKeyspaceSession.transaction().read()) {
-            boolean keyspaceExists = (tx.getAttributeType(KEYSPACE_RESOURCE.getValue()).attribute(keyspace) != null);
-            if (keyspaceExists) existingKeyspaces.add(keyspace);
-            return keyspaceExists;
-        }
-    }
-
-    public void deleteKeyspace(KeyspaceImpl keyspace) {
-        if (keyspace.equals(SYSTEM_KB_KEYSPACE)) {
-            throw GraknServerException.create("It is not possible to delete the Grakn system keyspace.");
-        }
-        deleteReferenceInSystemKeyspace(keyspace);
-    }
-
-    private void deleteReferenceInSystemKeyspace(KeyspaceImpl keyspace) {
-        try (TransactionOLTP tx = systemKeyspaceSession.transaction().write()) {
-            AttributeType<String> keyspaceName = tx.getSchemaConcept(KEYSPACE_RESOURCE);
-            Attribute<String> attribute = keyspaceName.attribute(keyspace.name());
-            if (attribute == null) {
-                throw GraknServerException.create("It is not possible to delete keyspace [" + keyspace.name() + "] as it does not exist.");
-            }
-            Thing thing = attribute.owner();
-            thing.delete();
-            attribute.delete();
-
-            existingKeyspaces.remove(keyspace);
-            tx.commit();
-        }
+    private final Cluster storage;
+    private final Set<String> internals = new HashSet<>(Arrays.asList("system_traces", "system", "system_distributed", "system_schema", "system_auth"));
+    
+    public KeyspaceManager(int storageNativeTransportPort) {
```

Comment:
can we inject the `Cluster` directly here? so that we can reuse this class in KGMS with no need to re-implementation

Reply:
Why can't we simply use it in KGMS as is? Is it because we will need to supply credentials?

Reply:
yes indeed

Reply:
fixed